core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -389,9 +371,28 @@ private void failAndRespond(Throwable cause, AggregatedHttpResponse res, Http2Er
         addCallbackAndFlush(cause, oldState, future);
     }
 
+    private void maybeLogFirstResponseBytesTransferred(ChannelFuture future) {
+        future.addListener((ChannelFuture f) -> {
+            if (f.isSuccess()) {
+                maybeLogFirstResponseBytesTransferred();
+            }
+        });
```

Comment:
Would it be possible to piggy-bag this into `addCallbackAndFlush()`, so that we keep the number of listeners minimal?

Reply:
Ah that's a good idea. I didn't put this logic in `addCallbackAndFlush` because this shouldn't be called after writing reset.
But, if it's resetting that means that `loggedResponseHeadersFirstBytesTransferred` is already set, so I didn't have to worry about it.